### ISSUE TITLE
Robot.txt, noindex, or similar

### ISSUE THREAD

**Issue by WalkerGriggs** [rwxrob/keg-spec#25]
One thing that jumped to mind after reading [Aggregation and attribution](https://github.com/rwxrob/keg-spec/blob/1a4889e89786bdfbc53efe2a0fd1e8707dc6d0df/5/README.md) is the potential for crawling, caching, etc.

I would love to follow other user's kegs, but in the off chance the authoring user doesn't want the content to be indexed or cached...

Do we think it's in the spirit of KEG to add some concept of a `noindex`? I think the dex node covers the use case of a robot file quite well.

It could either be part of the dex node (I'd guess that's the first place a scraper would land anyway) or another reserved, non-integer node.

**Comment by rwxrob** [comment]
This is a really good question. The foundation of KEG architecture depends heavily on caching to use it (the whole "no transport dependencies" thing). You have to get a copy of someone's keg somehow in order to even use it. But caching and creating an index are different things.

One reason to add this is to warn people that certain content is huge and not meant to be cached, but is linked locally somehow. Usually people who want this are planning on combining KEG as a layer on top of some other type of repository (machine learning models, source code, binaries, large videos). However, once content is within a KEG root directory it gets very hard to break it out. Over the KEG Web a crawler can follow links to create a mirrored copy of anything that isn't linked, but do we want to encourage people to mix noindex content with other indexed stuff?

The entire "don't crawl" thing is just a request on the Web with no real teeth at all. What you are describing is the reason I requested that people not quote from my zet without asking me because it is so live and sporadic and unfiltered. I still cannot stop people from doing it, but they might not even know otherwise. Giving a formal heads up about such things might be good.

I suppose I'm for adding something to the `keg` file since that is the first and only file any app needs to examine to get the preferences of the content creators and contains the `updated` field. Perhaps we could add a `noindex` or something like `indexing: none` or whatever. But we have to realized this is only a request. 

Publishing encrypt signed content is also an option for people who want more control over who reads it and when.

I think the best way to definitely answer this is to have a few solid examples of why someone would actually want it.

**Comment by WalkerGriggs** [comment]
> The entire "don't crawl" thing is just a request on the Web with no real teeth at all. 

Agreed. If anything, it's up to the crawler to respect those suggestions and there's no real consequence if they don't

> What you are describing is the reason I requested that people not quote from my zet without asking me because it is so live and sporadic and unfiltered.

This is more of what I was thinking. Less "I prefer my privacy" and more "my notes are subject to constant revision". One thing that comes to mind is a cache TTL, akin to DNS TTL.

Nodes that change often shouldn't be cached for long, whereas nodes that are almost never touched can be cached for a while. Nodes that have large videos or ML models would have a ttl of `0` or `nocache`.

In that sense, these are less of 'unenforceable requests' and more 'helpful suggestions' that'll keep catches relevant; especially when we consider each node as static.

I'd imagine this is a keg-wide value, which can be individually updated on nodes? Spitballing here, but completely agree that we'd need to collect use-cases from users.
